### ISSUE TITLE
Use VSCode Voice Support to allow Speech-To-Text

### ISSUE THREAD

**Issue by arturo-air** [continuedev/continue#3467]
### Validations

- [ ] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

The plugin does not accept voice typing, for the moment. I saw that @mkummer225 tried to implement it via Whisper #2047 , but closed this PR because he had to do the API connections.

### Solution

VSCode has added [Voice support](https://code.visualstudio.com/docs/editor/voice). I guess this makes it easier to implement Speech-To-Text.

**Comment by sestinj** [comment]
I worry that this would overly complicate the codebase while not satisfying the need for any IDE other than VS Code. If someone feels it can be done easily I'm potentially open to the contribution, but please let me know about your plan! Given that there are good OS-level STT tools, I think this isn't super urrgent

**Comment by arturo-air** [comment]
You are right, I didn't think there were more IDE using Continue than VSCode. I don't have time to investigate right now, so I'll close the issue, but I'll reopen it if I find some time to play with it.

**Comment by Symfomany** [comment]
Open setting in file or User setting to include API Keys in text to speech like OpenAI TTS,  Google Text To Speech or Amazon Alexa SDK, Librea Speech, Mozilla TTS , .. Maybe Mistral Voxstral for example https://mistral.ai/fr/news/voxtral
